### DISCUSSION TITLE
Como vocês organizam seus projetos pessoais?

### DISCUSSION THREAD

**Discussion by ortizvinicius** [frontendbr/forum#249] [Discussão]
Vou tentar ser objetivo: como vocês organizam seus projetos pessoais (de dev), isso inclui projetos open-source e fechados, linguagens e frameworks para aprender, artigos e livros para ler etc. Dos menores aos maiores.
Digo isso pois sei que todos somos ocupados e é difícil começar algo imediatamente e as vezes algo que iria te ajudar acaba se perdendo. Mas tenho certeza que os devs mais ativos já encontraram alguma solução prática e criativa para isso.

**Comment by woliveiras** [comment]
Senta que lá vem textão:

**Projetos pessoais Open ou Fechados, Linguagens e Frameworks**

Normalmente escrevo todas as ideias que tenho sobre o projeto em um caderno, em seguida vou eliminando as ideias que fugiram do foco do projeto (acontece bastante), depois tento selecionar as ideias e montar [MVPs](https://en.wikipedia.org/wiki/Minimum_viable_product). Isso só pra selecionar ideias que possam agregar ao projeto, pois nós podemos fugir bastante do foco quando estamos tendo aquele momento de inspiração. ;D

Com as ideias no papel, agora eu passo tudo para um board com o nome do projeto no [Trello](https://trello.com). Lá eu tento trabalhar com algo parecido com [Scrum](https://en.wikipedia.org/wiki/Scrum_%28software_development%29) que relato [aqui](https://woliveiras.com.br/posts/gerenciar-projetos-com-trello/).

No início do projeto vem as dúvidas de como fazer, o que usar, se começamos do zero ou não (como você levantou na issue)... Tudo depende.

Se for um projeto que precisa ser feito rapidamente para eu utilizar ou outras pessoas e é algo que precisa de Front End e Back End, então eu procuro alternativas rápidas como o uso de um [Boilerplate](https://github.com/woliveiras/kibe) para estruturar o projeto, alguma [plataforma fácil de trabalhar](https://firebase.google.com/) pra cuidar do meu Back End e ferramentas que agilizam as entregas como o [AngularJS](https://angularjs.org/) e gente para ajudar.

Já, se for algo que não precisa de pressa e é somente para uso pessoal, então eu procuro utilizar coisas que nunca mexi para aprender.

Hoje em dia, costumo trabalhar com o seguinte:
- AngularJS
- Stylus
- Gulp
- Algum BaaS

Mas, como disse, se der tempo eu mudo toda essa stack ai pra aprender algo novo!

Não costumo utilizar frameworks CSS, apesar de ter usado muito o Bootstrap e entender o uso deles. Prefiro fazer as coisas na mão e criar somente o que será necessário para o projeto em questão. Só uso alguma ferramenta como o [Kouto Swiss](http://kouto-swiss.io/) pra não precisar reescrever coisa que já existe ou já fiz em algum outro projeto e da pra reutilizar sem modificar o Design.

O mesmo acontece com o gulpfile. Costumo sempre iniciar um do zero para adequar ao projeto e as vezes até aprender a estruturar de forma diferente, como já aconteceu e eu compartilhei [aqui](https://github.com/woliveiras/modularize-your-gulpfile), conhecer novos plugins, etc.

**Sobre artigos e livros e coisas a aprender**

Eu peguei o costume de usar caderninho sempre que preciso planejar alguma coisa, então, mais uma vez, entra o caderno aqui também. :grimacing: 

Quando vou estudar alguma linguagem, framework ou algo novo (como tocar violão, andar de patins) eu busco o máximo de informações sobre o assunto. Procuro seguir pessoas que falam sobre isso, blogs e comunidades. Isso tira várias barreiras no aprendizado como a falta de locais para se informar.

Depois entra o caderninho: Escrevo num papel uma trilha de estudos parecida com a que fiz de [Front End](https://woliveiras.com.br/posts/guia-de-estudos-desenvolvedor-front-end-iniciante/) e tento seguir ela a risca. Faço como um sumário de livro e vou seguindo. Para fixar esse conhecimento eu escrevo em um caderno fazendo resumos do que aprendi e tentando escrever com as minhas palavras sobre o assunto, sem copiar do outro lugar, tento compartilhar o que aprendo no Blog pra que outras pessoas comentem sobre o assunto e me ensinem mais e tento discutir sobre o que estou aprendendo pelos foruns no mundão a fora.

Eu sou do tipo que gosta de estudar através de livros, mas nem todo mundo é assim, é importante descobrir qual seu estilo. E eu gosto de livros físicos, não de eooks, então tento sempre comprar uma cópia física do livro pra poder rabiscar, amassar, etc. 

Uso o [Feedly](http://feedly.com) para não perder nenhum artigo e as Redes Sociais pra me manter atualizado quanto a comunidade e tento me segurar pra não perder o foco e querer aprender tudo o que a comunidade mostra, pois esses caras tem o dom de mostrar coisas novas incríveis que sentimos vontade de começar a usar no outro dia! kkkk

Antigamente eu tinha o costume de aprender uma linguagem de programação (por cima) e uma distribuição Linux ou outro Sistema Operacional por ano pra pegar outros conceitos e outras maneiras de se fazer as coisas, mesmo eu sendo iniciante e não manjando tanto - Esse ano não consegui priorizar isso por conta dos objetivos da vida (comecei a faculdade e estou me lascando).

Foi assim que eu conheci o PHP, um pouco de Ruby, um pouco de Python, um pouco de Java (uuurrgh), o basicão de NodeJS e coisas como IoT e Big Data. Isso me ajudou muito a aprender a pesquisar no Google, pois cada ferramenta da um problema diferente e a pessoa precisa se virar nos trinta pra resolver.

Em todos os casos, o caderninho é o que sempre me ajuda a remover coisas do caminho e pensar bem o que vou fazer e como fazer. Fiz um [post](https://woliveiras.com.br/posts/como-alcancar-objetivos-rapidamente-nos-estudos/) sobre estudos e algumas coisas que falei aqui.

É algo assim mesmo que você queria @ortizvinicius?

**Comment by ortizvinicius** [comment]
Cara, perfeito.
Já me cadastrei no trello, muito bom.
Muito obrigado pela resposta.

**Comment by woliveiras** [comment]
\o/

É nozes mano

**Comment by rubenmarcus** [comment]
Opa vale a pena conhecer o [Wrike ](http://www.wrike.com)  tambem, para projetos pessoais é free.

**Comment by felipefialho** [comment]
@woliveiras Tem twitter?

**Comment by woliveiras** [comment]
Sim sim @LFeh, https://twitter.com/w_oliveiras

**Comment by adeonir** [comment]
@woliveiras, textão hein!? mas muito bom.

**Comment by camilatigre** [comment]
muito bom @woliveiras ! bem organizado e god bless o caderinho haha

**Comment by daniloagostinho** [comment]
@ortizvinicius Crie uma lista de links interessantes para me manter atualizado no mundo da Tecnologia e  organizar o fluxo de links que vejo. :tada:  Se quiser da uma olhada nele pode acessar [aqui](https://github.com/DaniloAgostinho/update). 

@woliveiras  Sou a favor do caderninho também **\\o//** mais confesso que preciso criar o hábito de escrever mais.... 

Achei super interessante a ideia de aprender **IoT**, **Big Data**. esta na minha lista de coisas pra aprender.

**Comment by MaizerGomes** [comment]
Bem, eu sigo +/- o que o @woliveiras falou aí em cima:

Começo do bloco de notas e caneta e depois passo para o [trello ](https://trello.com)criando uma board com o nome do projecto e várias listas que possuem as diversas tarefas que vou precisar executar em forma de módulos e até mesmo as tecnologias que preciso aprender, se for o caso.

Muitas vezes surgem ideias muito boas sobre o projecto enquanto estou em casa da namorada, no restaurante ou então durante o sonho (sim, durante o sonho.. 😪 ) então ando sempre com um bloquinho por perto para qualquer coisa.

Mas sendo sincero, não sou muito de planear as coisas então vou fazendo aquilo que me é mais fácil de terminar. Algumas vezes já me arrependi disso mas em outras levei tanto tempo a planear que acabei perdendo interesse no projecto. Então vou com a onda do momento. 😆 

A minha forma de aprender é via vídeo aulas e blogs onde vou buscar sempre me actualizar sobre as melhores práticas nas ferramentas que vou usar.
Raramente leio ebooks e não me lembro de alguma vez ter lido um livro físico para aprender uma linguagem, o que se deve principalmente à grande dificuldade de encontrar livros sobre informática no meu país ([Moçambique](https://pt.wikipedia.org/wiki/Mo%C3%A7ambique)) 😞 .


Sempre crio uma nova pasta de bookmarks no meu navegador, dou nome do projecto e vou favoritanto sempre que encontro uma página com algo que acho que posso usar no projecto. Por vezes acabo com uma lista gigante porque muitos deles leio só o primeiro parágrafo e favorito com o intuito de ler quando chegar em casa, mas acabo não lendo. Mas pelo menos sei que está lá para quando precisar.

Sempre uso um repositório [git ](https://git-scm.com/)para gerir o versionamento do meu projecto e hospedo no [bitbucket.com](https://bitbucket.org/).
Divido as etapas do desenvolvimento em algumas versões (tags) e uso uma branch separada para cada versão que tem alterações significantes (master/develop/v1.0/v2.0/v3.0).

Ah! Como eu não tenho muito domínio do front-end, acabo sempre iniciando com um boilerplate e cozinho-o até chegar a um ponto mais personalizado.

Não sei se cheguei ao ponto mas para não ficar longo demais, vou parar por aqui.

**Comment by matheusmpessoa** [comment]
@woliveiras quais perfis de tecnologia você segue no Feedly e recomenda?

**Comment by woliveiras** [comment]
Opa, melhor que falar é mandar esse arquivinho aqui @matheusmpessoa https://github.com/woliveiras/dotfiles/blob/master/others/feedly.opml

Só baixar e importar no Feedly que tem site pra caramba!!! 😄 

Todos os dias tem mais de 200 artigos na caixa de entrada. Muita informação.

**Comment by herberthenrique** [comment]
Primeiro, eu tenho um [Bullet Journal](http://bulletjournal.com/) onde tento organizar um pouco da minha vida pessoal e um macro dessas coisas.

Para estudo, lá eu tenho uma lista de "coisas" que quero aprender e uma ordem pra cada.
Baseado nisso, eu tenho uma pasta no favoritos do chrome com cursos/etc que to fazendo e um aviso no calendário do google para passar 2h de segunda/quarta/sexta a noite estudando. Como costumo fazer um esporte (boxe) nesses dias antes acabo ficando com a mente mais limpa pra isso.

Para projetos pessoais que envolvem programar, eu tenho uma outra lista no journal e organizo suas pendências com issues (seja no github para open-sources ou gitlab para algo mais fechado).

Basicamente, meu journal tem me salvado nos últimos meses :D

**Comment by raulfernando** [comment]
@giioohbernini
